rotkehlchen/chain/evm/decoding/base.py

```diff
@@ -141,14 +146,15 @@ def decode_erc20_721_transfer(
             return None
 
         extra_data = None
-        event_type, location_label, counterparty, verb = direction_result
+        event_type, location_label, address, counterparty, verb = direction_result
+        notes_counterparty = counterparty or address  # counterparty if found, otherwise address
```

Comment:
Good naming is better than comments

```suggestion
        counterparty_or_address = counterparty or address
```